app/helpers/pagelets_helper.rb

```diff
@@ -1,6 +1,13 @@
 module PageletsHelper
+  def virtual_path
+    @virtual_path
+  end
+
   def pagelets_for(mountpoint)
-    Pagelets::Manager.sorted_pagelets_at("#{controller_name}/#{action_name}", mountpoint)
+    result = ["#{controller_name}/#{action_name}", virtual_path].map do |key|
```

Comment:
When called from a regular template, outside of of a partial, `"#{controller_name}/#{action_name}"` and `virtual_path` both return the same string, causing one pagelet registered at say, `hosts/new` to be returned twice from this method.

Consider adding a `.uniq` in here, plus a unit test.